### PR TITLE
Clean up HMI synoptic map example

### PR DESCRIPTION
**Pull request by dstansby**
I think this example started off very complicated (https://github.com/sunpy/sunpy/pull/1809), but has got simpler and simpler over the years as more stuff is done automatically by sunpy or WCSAxes. This PR cleans up the example to

- Make sure the comments are correct
- Remove redundant code

### CODE REVIEW

File: examples/plotting/hmi_synoptic_maps.py [hunk 1]
```diff
@@ -1,12 +1,9 @@
 """
-==============================================
-Loading an HMI daily synoptic image into a Map
-==============================================
-
-In this example we load the Daily Synoptic Maps produced by the HMI team. This
-data is an interesting demonstration of sunpy's Map class as it is not in the
-more common Helioprojective coordinate system, it is in Heliographic Carrington
-coordinates and in a non-trivial Cylindrical Equal Area projection.
```
Comment (nabobalis) [inline examples/plotting/hmi_synoptic_maps.py:9]: I feel like this provides some more context to the example?

File: examples/plotting/hmi_synoptic_maps.py [hunk 2]
```diff
@@ -15,49 +12,36 @@
...
-cb.set_label("LOS Magnetic Field [Gauss]")
+cb.set_label("Radial magnetic field [Gauss]")
 
-# Another horrible hack to make the ticks draw on the RHS
-axes.set_ylim((1, syn_map.data.shape[0] - 1))
-plt.title("{} {}-{}".format(syn_map.meta['content'], syn_map.meta['CAR_ROT'],
-                            syn_map.meta['CAR_ROT'] + 1))
+# In order to make the x-axis ticks show, the bottom y-limit has to be adjusted slightly
+axes.set_ylim(bottom=0)
+axes.set_title("{} {}-{}".format(syn_map.meta['content'], syn_map.meta['CAR_ROT'],
```
Comment (nabobalis) [inline examples/plotting/hmi_synoptic_maps.py:44]: f string?

File: examples/plotting/hmi_synoptic_maps.py (hunk 1 above)
Comment (dstansby) [reply examples/plotting/hmi_synoptic_maps.py:9]: I didn't think it was that relevant any more as `Map` works out of the box for different projections. Happy to put it back though?

Comment (nabobalis) [reply examples/plotting/hmi_synoptic_maps.py:9]: That's fair but I think leaving the example main text as `In this example we load a synoptic map produced by the HMI team.` is a bit lacking on information or context.

Comment (dstansby) [reply examples/plotting/hmi_synoptic_maps.py:9]: I put this back in, with some minor modifications.

**Review by nabobalis — APPROVED** [review]

**Pull request merged**